### PR TITLE
BDRSPS-110 Implement SSD metadata mapping

### PR DESCRIPTION
**Pull request by joecrowleygaia**
First iteration of the systematic survey metadata abis mapping. Design is based largely on the supplied graph diagram and excludes the grey dashed bordered boxes it contained.

### CODE REVIEW

File: abis_mapping/utils/rdf.py [hunk 1]
```diff
@@ -31,6 +31,7 @@
     ("geo", namespaces.GEO),
     ("tern", namespaces.TERN),
     ("dwc", namespaces.DWC),
+    ("bdra", namespaces.BDRA),
```
Comment (serge-gaia) [inline abis_mapping/utils/rdf.py:34]: @joecrowleygaia , could change the namespace to be just bdr/BDR (not BDRA). I've told Mieke to change it but it must have slipped her mind.

**Review by serge-gaia — CHANGES_REQUESTED** [review]

File: abis_mapping/utils/rdf.py (hunk 1 above)
Comment (joecrowleygaia) [reply abis_mapping/utils/rdf.py:34]: @serge-gaia should i be changing the uri of the namespace from `http://example.com/bdra-schema/` to `http://example.com/bdr-schema/`

Comment (serge-gaia) [reply abis_mapping/utils/rdf.py:34]: yes

Comment (serge-gaia) [reply abis_mapping/utils/rdf.py]: It's just a placeholder namespace that we, Gaia, use when we make things up. For example a new Class that doesn't exist in the Tern schema but that we think should exist.

Comment (joecrowleygaia) [reply abis_mapping/utils/rdf.py]: changed both URI and prefix lettering

File: abis_mapping/templates/survey_metadata/mapping.py [hunk 2]
```diff
@@ -57,7 +74,260 @@ def apply_mapping(
...
+            # Map row
+            self.apply_mapping_row(
+                row=row,
+                dataset=dataset_iri,
+                graph=graph,
+                base_iri=base_iri,
+            )
+
+            # Check whether to yield a chunk
+            if utils.chunking.should_chunk(row, rows, chunk_size):
```
[the remark this answers is not in the document]
Comment (joecrowleygaia) [reply abis_mapping/templates/survey_metadata/mapping.py]: removed chunking and moved "chunk_size" argument of the occurrence mapping methods to be a **kwarg to make the base mapping method suit this case.

**Review by serge-gaia — CHANGES_REQUESTED** [review]

File: abis_mapping/templates/survey_metadata/mapping.py [hunk 3]
```diff
@@ -53,11 +70,252 @@ def apply_validation(
...
+        """Adds the BDR survey to the graph.
+
+        Args:
+            uri (rdflib.URIRef): URI of the survey.
+            dataset (rdflib.URIRef): URI of the dataset.
+            graph (rdflib.Graph): The graph to be modified.
+        """
+        # Add type and dataset
+        graph.add((uri, a, utils.namespaces.BDR.Survey))
+        graph.add((uri, rdflib.VOID.inDataset, dataset))
```
[the remark this answers is not in the document]
Comment (joecrowleygaia) [reply abis_mapping/templates/survey_metadata/mapping.py:245]: According to her corresponding .ttl however the connection is made in several locations. It was a bit hard to know which to believe and which to ignore. Would you like me to remove all dataset connections except the one made with the Project main node?

[the remark this answers is not in the document]
Comment (joecrowleygaia) [reply abis_mapping/templates/survey_metadata/mapping.py:245]: removing

[the remark this answers is not in the document]
Comment (serge-gaia) [reply abis_mapping/templates/survey_metadata/mapping.py:245]: yes I prefer removing them.

[the remark this answers is not in the document]
Comment (joecrowleygaia) [reply abis_mapping/templates/survey_metadata/mapping.py]: removed dataset predicate/objects from the survey and survey methodology nodes. They should now only appear at the project main node.

**Review by serge-gaia — APPROVED** [review]

**Pull request merged**